gtda/homology/simplicial.py

```diff
@@ -720,3 +721,230 @@ def plot(Xt, sample=0, homology_dimensions=None):
+        possible precision.
+
+    max_edge_length : float, optional, default: ``numpy.inf``
+        Upper bound on the maximum value of the filtration parameter. Points
+        whose distance is greater than this value will never be connected by
+        an edge, and topological features at scales larger than this value
+        will not be detected.
+
+    infinity_values : float or None, default : ``None``
+        Which death value to assign to features which are still alive at
+        filtration value `max_edge_length`. ``None`` means that this
+        death value is declared to be equal to `max_edge_length`.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    infinity_values_ : float
+        Effective death value to assign to features which are still alive at
+        filtration value `max_edge_length`.
+
+    See also
+    --------
+    VietorisRipsPersistence, SparseRipsPersistence, EuclideanCechPersistence, \
+    ConsistentRescaling, ConsecutiveRescaling
+
+    Notes
+    -----
+    The `pyflagser <https://github.com/giotto-ai/pyflagser>`_ Python package
+    is used for binding `Flagser <https://github.com/luetge/flagser>`_, a C++
+    backend for computing the (persistent) homology of (filtered) directed
+    flag complexes.
+
+    Persistence diagrams produced by this class must be interpreted with
+    care due to the presence of padding triples which carry no information.
+    See :meth:`transform` for additional information.
+
+    References
+    ----------
+    [1] D. Luetgehetmann, D. Govc, J. P. Smith, and R. Levi, "Computing \
+        persistent homology of directed flag complexes", Algorithms,
+        13(1), 2020.
+
+    """
+
+    _hyperparameters = {
+        'homology_dimensions': {
+            'type': (list, tuple), 'of': {
+                'type': int, 'in': Interval(0, np.inf, closed='left')}},
+        'directed': {'type': bool},
+        'coeff': {'type': int, 'in': Interval(2, np.inf, closed='left')},
+        'max_entries': {'type': int},
+        'max_edge_length': {'type': Real},
+        'infinity_values': {'type': (Real, type(None))}
+    }
+
+    def __init__(self, homology_dimensions=(0, 1), directed=True, coeff=2,
+                 max_entries=-1, max_edge_length=np.inf, infinity_values=None,
+                 n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.directed = directed
+        self.coeff = coeff
+        self.max_entries = max_entries
+        self.max_edge_length = max_edge_length
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _flagser_diagram(self, X):
+        X[X >= self.max_edge_length] = self.max_edge_length
+
+        Xdgms = flagser(X[X[:, 0] != np.inf],
```

Comment:
What is `X[:, 0] != np.inf` and why do we exclude those rows?

Reply:
There is a historical reason for this that maybe @gtauzin remembers but I can't. I think it had to do with the implementation of some of the classes in the images submodule. Perhaps `ImageToPointCloud` which used to return arrays with some infinities in them?

Reply:
(It is the same almost everywhere else in `gtda.homology`)

Reply:
It seems quite dangerous, when (and only when!) `metric='precomputed'`, in both `VietorisRipsPersistence` and `FlagserPersistence`.
Take the following example:
```
Xs = np.array([[[0, 1, np.inf, 1],
              [1, 0, 1, 10],
              [np.inf, 1, 0, 1],
              [1, 10, 1, 0]]]).astype(float)
```
Then, Flagser works if we remove `X[:, 0] != np.inf`. Otherwise, it removes one row from the distance matrix and you get something not square  - strangely, the shape of the input is not validated, and the kernel crashes.

VietorisRips, in its current state, behaves as Flagser. But, it does not handle`np.inf`, so we may want to find another solution

Reply:
Apparently whatever reason there ever was is no longer needed. Good, because it really shouldn't be there. I'll remove it. Though I didn't quite understand your latest message.

Reply:
I am just pointing out that if you remove the 'input masking' in ripser as well, VietorisRipsPersistence will fail. Maybe `np.inf` in the distance matrix is a valid reason for it to fail.

Reply:
@wreise am I right to believe that this discussion can be resolved following #351 ?